### PR TITLE
[3.3.5] Scripts/Spells: handle auras on unequipping scripted items

### PR DESCRIPTION
**Pull request by DanVS**
**Changes proposed:**
When some scripted items (capacitor-like) are unequiped, remove their auras from owner. List of handled items:
- [Tiny Abomination in a Jar](http://www.wowhead.com/item=50351)
- [Shadowmourne](http://www.wowhead.com/item=49623)
- [Lightning Capacitor](http://www.wowhead.com/item=28785)
- [Thunder Capacitor](http://www.wowhead.com/item=38072)
- [Reign of the Dead](http://www.wowhead.com/item=47316)

Not sure that this is complete list, so help here would be appreciated.

**Target branch(es):** 3.3.5/master
- [x] 3.3.5
- [ ] master

**Issues addressed:** Ref #20415

**Tests performed:** Builds, tested in-game.

### CODE REVIEW

File: src/server/scripts/Spells/spell_item.cpp [hunk 1]
```diff
@@ -1785,6 +1791,11 @@ class spell_item_shadowmourne : public AuraScript
         }
     }
 
+    void OnRemove(AuraEffect const*, AuraEffectHandleModes)
```
Comment (ccrs) [inline src/server/scripts/Spells/spell_item.cpp]: unregistered

File: src/server/scripts/Spells/spell_item.cpp [hunk 2]
```diff
@@ -256,9 +256,15 @@ class spell_item_anger_capacitor : public SpellScriptLoader
                 caster->CastSpell(target, spellId, aurEff);
             }
 
+            void OnRemove(AuraEffect const*, AuraEffectHandleModes)
+            {
+            	GetTarget()->RemoveAurasDueToSpell(SPELL_MOTE_OF_ANGER);
```
Comment (ccrs) [inline src/server/scripts/Spells/spell_item.cpp]: is target guaranteed on ``AfterEffectRemove`` ?

Comment (DanVS) [reply src/server/scripts/Spells/spell_item.cpp]: Could you tell me what do you mean with that?

Comment (ccrs) [reply src/server/scripts/Spells/spell_item.cpp]: I dont know if the return value of GetTarget at that stage of the aura effect lifespan, is guaranteed to be the actual target of the aura or just nullptr. Do you?

Comment (DanVS) [reply src/server/scripts/Spells/spell_item.cpp]: Isn't [this](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Spells/Auras/SpellAuras.cpp#L56) enough?

File: src/server/scripts/Spells/spell_item.cpp (hunk 1 above)
Comment (DanVS) [reply src/server/scripts/Spells/spell_item.cpp]: Fixed.

File: src/server/scripts/Spells/spell_item.cpp (hunk 2 above)
Comment (ccrs) [reply src/server/scripts/Spells/spell_item.cpp]: nope, unrelated

Comment (DanVS) [reply src/server/scripts/Spells/spell_item.cpp]: OK, I will add corresponding check.

**Comment by DanVS** [comment]
Fixed, I guess.

File: src/server/scripts/Spells/spell_item.cpp (hunk 2 above)
Comment (ccrs) [reply src/server/scripts/Spells/spell_item.cpp]: I'm not saying that its needed, I'm saying that I dont know if its needed.

File: src/server/scripts/Spells/spell_item.cpp [hunk 3]
```diff
@@ -256,9 +256,16 @@ class spell_item_anger_capacitor : public SpellScriptLoader
                 caster->CastSpell(target, spellId, aurEff);
             }
 
+            void OnRemove(AuraEffect const*, AuraEffectHandleModes)
+            {
+            	if (GetTarget())
```
Comment (ccrs) [inline src/server/scripts/Spells/spell_item.cpp]: I'm not saying that its needed, I'm saying that I dont know if its needed.

Comment (DanVS) [reply src/server/scripts/Spells/spell_item.cpp]: Well, in [SpellScript](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Spells/SpellScript.h#L922) check is not needed.

File: src/server/scripts/Spells/spell_item.cpp [hunk 4]
```diff
@@ -3456,9 +3470,16 @@ class spell_item_trinket_stack : public SpellScriptLoader
                 caster->CastSpell(target, _triggerSpell, aurEff);
         }
 
+        void OnRemove(AuraEffect const*, AuraEffectHandleModes)
+        {
+        	if (GetTarget())
+       			GetTarget()->RemoveAurasDueToSpell(_stackSpell);
```
Comment (MitchesD) [inline src/server/scripts/Spells/spell_item.cpp]: seems like tabs or wrong spaces

Comment (tkrokli) [reply src/server/scripts/Spells/spell_item.cpp]: Yup, @DanVS needs to configure editor to auto convert tabs to 4 spaces.

**Comment by DanVS** [comment]
Woops :\ Sorry for dirtiness.

**Comment by Keader** [comment]
lul, need remove unrelated commits xD

**Comment by tkrokli** [comment]
^^ rebase on current `3.3.5` branch to get rid of those SQL updates already in TC ?

**Comment by DanVS** [comment]
Probably.

**Comment by DanVS** [comment]
@tkrokli, could you please rebase this to remove old commits? :)

**Comment by tkrokli** [comment]
I should be able to. Just have to fork your branch first.

**Comment by Shauren** [comment]
`if (GetTarget())` is unneeded in AfterEffectRemove hooks

File: src/server/scripts/Spells/spell_item.cpp (hunk 1 above)
Comment (Keader) [reply src/server/scripts/Spells/spell_item.cpp]: weird param :(

Comment (DanVS) [reply src/server/scripts/Spells/spell_item.cpp]: You mean comments? `AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/`

Comment (Keader) [reply src/server/scripts/Spells/spell_item.cpp]: y

**Comment by Keader** [comment]
Thanks

**Pull request merged**